### ISSUE TITLE
Popover accessibility: Navigate calendar items with keyboard

### ISSUE THREAD

**Issue by catlabs** [fullcalendar/fullcalendar#6624]
### Checklist

Please mark these items with an [x]

  - [x] I've already searched through [existing tickets](https://fullcalendar.io/issues)
  - [x] Other people will find this feature useful

### Feature Description

(+ x More) is now indeed selectable with keyboard and open the popover (v5.10.0) but when popover is opened, we cannot navigate through events inside of this one.

**Comment by MagdalenaHenke** [comment]
Adding some context on this issue: 

**The elements in the popover are keyboard accessible, but they are in the wrong place in the html structure, so they aren't the next thing tabbed to.**

The popover gets appended _after_ the table component, which means that to tab to the events in the popover, one has to tab through / past all other events in the calendar, reach the end of all visible events in the calendar table, and _then_ goes through the popover events (since tab order follows html order).

<img width="1434" alt="Screen Shot 2023-03-31 at 2 51 42 AM" src="https://user-images.githubusercontent.com/6024274/229045201-dbc48458-e2fa-44c7-86fa-4868efe820e6.png">

What should happen instead is having the popover events be in the tab order between the events of the day that the popover belongs to, and any events after that - the popover events should enter the tab order where the popover visually is added to the page.

This can be fixed by inserting the popover html after the `+x` link, instead of after the table component, or via using other methods to manually manage focus order. 


```
"@fullcalendar/core": "^6.1.4",
"@fullcalendar/daygrid": "^6.1.4",
"@fullcalendar/react": "^6.1.4",
```

**Comment by arshaw** [comment]
Thank you for the very thorough recommendation @MagdalenaHenke. Avoiding using Portals is one solution, but FWIW, react-area uses [a different solution](https://react-spectrum.adobe.com/react-aria/Popover.html#popover-2) where it focuses the popover element, which allows the next TAB to go into the focusable elements within the popover, and if the user presses ESC to close the popover, focus returns to the element that opened it. I still must research the subtleties and drawbacks of this approach.

**Comment by MagdalenaHenke** [comment]
Absolutely, makes sense! There are multiple ways this could be addressed!

**Comment by arshaw** [comment]
This has been implemented in [v7.0.0-beta.4](https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-beta.4)

In this demo, tab all the way to +more link, press Enter or Space to open it, continue tabbing through the events, and as you reach the end, focus is restored to the +more link. You can then continue tabbing through items after the +more link.
https://codepen.io/arshaw/pen/mybXMmz?editors=0010
